StackExchange.Redis/StackExchange/Redis/ConnectionMultiplexer.cs

```diff
@@ -37,13 +37,16 @@ public static Task<T> ObserveErrors<T>(this Task<T> task)
             return task;
         }
 
+        [ThreadStatic]
+        public static bool ContinueOnCapturedContext;
```

Comment:
I'm not sure what you mean re a custom sync-context; the sync-context
relates to the calling infrastructure : we shouldn't be messing with it at
all, unless we're genuinely providing the threading model (which: we aren't)

On Mon, 16 May 2016 20:29 Brandon, <notifications@github.com> wrote:

> In StackExchange.Redis/StackExchange/Redis/ConnectionMultiplexer.cs
> <https://github.com/StackExchange/StackExchange.Redis/pull/264#discussion_r63410354>
> :
>
> > @@ -37,13 +37,16 @@ public static Task<T> ObserveErrors<T>(this Task<T> task)
> >              return task;
> >          }
> >
> > +        [ThreadStatic]
> > +        public static bool ContinueOnCapturedContext;
>
> I used this technique to avoid having to re-factory the existing code too
> much. SynchronizationContext.Current is also thread-local so it seems to
> make sense to me. I suppose ForAwait can change to something like
>
> public static ConfiguredTaskAwaitable ForAwait(this Task task)
> {
>     var useContext = SynchronizationContext.Current.GetType() == typeof(SychronousSynchronizationContext);
>     return task.ConfigureAwait(useContext);
> }
>
> Alternatively we could pass true/false to every async method.
>
> How do you feel about using a custom SyncrhonizationContext implementation in general? I chose this method because it required the least amount of code change and allowed the use of async/await to continue.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/StackExchange/StackExchange.Redis/pull/264/files/8e350dfae25c7947a9e6330774c64a477eedf088#r63410354>
>